examples/air/air.py

```diff
@@ -118,13 +119,25 @@ def local_model(self, n, batch=None, **kwargs):
         z_where = []
 
         for t in range(self.num_steps):
-            state = self.model_step(t, n, state, batch, **kwargs)
+            state = self.model_step(t, n, state, **kwargs)
             z_where.append(state.z_where)
             z_pres.append(state.z_pres)
 
         return (z_where, z_pres), state.x
 
-    def model_step(self, t, n, prev, batch, z_pres_prior_p=default_z_pres_prior_p):
+    def model(self, data, _, **kwargs):
+        pyro.module("decode", self.decode)
+        with pyro.iarange('data', data.size(0), use_cuda=self.use_cuda) as ix:
+            batch = data[ix]
+            n = batch.size(0)
+            (z_where, z_pres), x = self.prior(n, **kwargs)
+            pyro.observe('obs',
+                         dist.normal,
+                         batch.view(n, -1),
+                         x.view(n, -1),
+                         (self.likelihood_sd * self.ng_ones(1)).expand(n, self.x_size ** 2))
+
+    def model_step(self, t, n, prev, z_pres_prior_p=default_z_pres_prior_p):
 
         # Sample presence indicators.
         if not self.fudge_z_pres:
```

Comment:
some of the trickier things could perhaps use more comments? e.g. fudge_z_pres

Reply:
I don't think it's worth keeping `fudge_z_pres` around, so I'll remove that in a follow up PR. I'll check for similar things at the same time, but let me know if there's anything else that needs specific attention.